tests/media/server/main/mediaPipeline/MiscellaneousFunctionsTest.cpp

```diff
@@ -320,6 +320,71 @@ TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, GetVolumeSuccess)
     EXPECT_EQ(resultVolume, kCurrentVolume);
 }
 
+/**
+ * Test that SetMute returns failure if the gstreamer player is not initialized
+ */
+TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, SetMuteFailureDueToUninitializedPlayer)
+{
+    const bool kMute{false};
+    mainThreadWillEnqueueTaskAndWait();
+    EXPECT_FALSE(m_mediaPipeline->setMute(kMute));
+}
+
+/**
+ * Test that SetMute returns failure if the gstreamer player is not initialized
+ */
+TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, SetMuteSuccess)
+{
+    const bool kMute{false};
+    loadGstPlayer();
+    mainThreadWillEnqueueTaskAndWait();
+
+    EXPECT_CALL(*m_gstPlayerMock, setMute(kMute));
+    EXPECT_TRUE(m_mediaPipeline->setMute(kMute));
+}
+
+/**
+ * Test that GetMute returns failure if the gstreamer player is not initialized
+ */
+TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, GetMuteFailureDueToUninitializedPlayer)
+{
+    mainThreadWillEnqueueTaskAndWait();
+    bool resultMute{};
+    EXPECT_FALSE(m_mediaPipeline->getMute(resultMute));
+}
+
+/**
+ * Test that GetMute returns failure if the gstreamer API fails
+ */
+TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, GetMuteFailure)
+{
+    loadGstPlayer();
+    mainThreadWillEnqueueTaskAndWait();
+    bool resultMute{};
+    EXPECT_CALL(*m_gstPlayerMock, getMute(_)).WillOnce(Return(false));
+    EXPECT_FALSE(m_mediaPipeline->getMute(resultMute));
+}
+
+/**
+ * Test that GetMute returns success if the gstreamer API succeeds and gets mute
+ */
+TEST_F(RialtoServerMediaPipelineMiscellaneousFunctionsTest, GetMuteSuccess)
+{
+    constexpr bool kCurrentMute{false};
+    bool resultMute{};
+
+    loadGstPlayer();
+    mainThreadWillEnqueueTaskAndWait();
+    EXPECT_CALL(*m_gstPlayerMock, getMute(_))
+        .WillOnce(Invoke(
```

Comment:
Not wrong and don't need to change it, but IMO DoAll(SetArgReferee<0>(kCurrentMute), Return(true)) is better here, but its personal preference.